### ISSUE TITLE
System stats

### ISSUE THREAD

**Issue by stephendade** [stephendade/Rpanion-server#31]
Add warning when system temperature, voltage or averaged CPU exceeds limits.

This would allow users to diagnose some stability issues, especially when running video streaming

**Comment by lorenzing** [comment]
Great function! I think we can also integrate the reading of undervoltage and cpu throtteling.
```
vcgencmd get_throttled
vcgencmd measure_temp
vcgencmd measure_volts core
vcgencmd measure_clock core
```
The `vcgencmd get_throttled` output bits are:
![clipboard_eddfc4bfa4efe78b914acf08a906d9f3f](https://user-images.githubusercontent.com/10693529/151119826-18fbed48-cc84-4523-a63d-4b5af6e4f8fa.png)

Other commands can be integrated, such as reading the gpu or free ram.
`vcgencmd get_mem arm/gpu`

For example this is the dashboard of my Raspberry Pi (Zero 2 W) which controls the 3D printer:
![Schermata 2022-01-26 alle 08 30 07](https://user-images.githubusercontent.com/10693529/151121169-bfb2a792-c8de-4286-b33c-1d9cb3fe4063.png)

**Comment by stephendade** [comment]
I'd prefer a more generic approach, as Rpanion-server also runs on Jetsons and a few x86 platforms now. Is there a more generic functions we can use?

**Comment by lorenzing** [comment]
Hello Stephen, sorry for the delay in replying. You are absolutely right about the compatibility of the feature with all supported platforms. However, I believe it is important to know if the raspberry get throttled.

**Comment by stephendade** [comment]
> However, I believe it is important to know if the raspberry get throttled.

Yes, that's fine. I'm generally happy to put in small amounts of Pi-specific functionality

**Comment by ddd999** [comment]
lm-sensors is probably the most widely supported way to get hardware monitoring info: https://github.com/lm-sensors/lm-sensors
